### ISSUE TITLE
[WebAPI] create customer account via WebAPI still need email confirmation

### ISSUE THREAD

**Issue by wubinworks** [magento/magento2#39254]
### Preconditions and environment

- Magento 2.4.6 & 2.4.7
- Set config path [`customer/create_account/confirm`](https://github.com/magento/magento2/blob/a4cf5e62468770fdb9f7b48938f6644ae354a51b/app/code/Magento/Customer/etc/adminhtml/system.xml#L110) to `1` and clear cache to let this setting take effect

### Steps to reproduce

Create a customer account via WebAPI as described in [official document](https://developer.adobe.com/commerce/webapi/rest/tutorials/orders/order-create-customer/).

### Expected result

No confirmation required if customer account is created via WebAPI.

### Actual result

Try login on frontend and you will see a "This account is not confirmed" message.
Verify the created customer in `customer_entity` table, the `confirmation` field is not `NULL`.

### Additional information

BUT, check [AccountManagementApi.php](https://github.com/magento/magento2/blob/a4cf5e62468770fdb9f7b48938f6644ae354a51b/app/code/Magento/Customer/Model/AccountManagementApi.php#L139) and [di.xml](https://github.com/magento/magento2/blob/a4cf5e62468770fdb9f7b48938f6644ae354a51b/app/code/Magento/Customer/etc/webapi_rest/di.xml#L37)
```
    public function createAccount(CustomerInterface $customer, $password = null, $redirectUrl = '')
    {
        $this->validateCustomerRequest($customer);
        $customer = parent::createAccount($customer, $password, $redirectUrl);
        $customer->setConfirmation(null);
        // Forgot to save customer
        return $customer;
    }
```
From the above code, the `confirmation` should be `NULL` if account is created via WebAPI even `customer/create_account/confirm` is set to `1`.
<ins>**But this code forgot to save customer after setting confirmation to `null`.**</ins>

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [X] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
Hi @wubinworks,

Thanks for your reporting and collaboration.

We have verified the issue in Latest 2.4-develop instance and the issue is not reproducible.Kindly refer the screenshots.

<img width="884" alt="Screenshot 2024-10-21 at 11 08 47" src="https://github.com/user-attachments/assets/eef31e62-66d5-4905-99e2-f2c6167d3c9d">



<img width="1333" alt="Screenshot 2024-10-21 at 11 06 49" src="https://github.com/user-attachments/assets/b819ee47-3df8-4a12-a2a4-ac04f86012bc">


<img width="1333" alt="Screenshot 2024-10-21 at 11 02 33" src="https://github.com/user-attachments/assets/e64696ad-08c9-4489-8585-30c7ed977180">


<img width="804" alt="Screenshot 2024-10-21 at 11 09 26" src="https://github.com/user-attachments/assets/64364395-8356-4a69-9603-e69b19069405">



It seems to be an expected behaviour of Magento.we have created customer via Webapi and still `confirmation` field having value in `customer_entity` table.Hence we have to confirm the account before logging into the customer.Kindly let us know if you are still facing any issue.

Thanks.
